src/openms/source/ANALYSIS/ID/FIAMSDataProcessor.cpp

```diff
@@ -0,0 +1,422 @@
+  filter.filter(spectrum);
+
+  MorphologicalFilter morph_filter;
+  morph_filter.filter(spectrum);
+
+  PeakPickerHiRes picker;
+
+  MSSpectrum picked;
+  picker.pick(spectrum, picked);
+
+  return picked;
+}
+
+FeatureMap FIAMSDataProcessor::convertToFeatureMap(const MSSpectrum & input) {
+  FeatureMap output;
+  for (auto it = input.begin(); it != input.end(); ++it) {
+      Feature f;
+      f.setIntensity(it->getIntensity());
+      f.setMZ(it->getMZ());
+      output.push_back(f);
+      output[0].setMetaValue("scan_polarity", polarity_);
+  }
+  return output;
+}
+
+void FIAMSDataProcessor::runAccurateMassSearch(FeatureMap & input, OpenMS::MzTab & output) {
+  Param ams_param;
+  ams_param.setValue("ionization_mode", "auto");
+  ams_param.setValue("mass_error_value", 1e+06 / (resolution_*2));
+  ams_param.setValue("db:mapping", ListUtils::create<String>(db_mapping_));
+  ams_param.setValue("db:struct", ListUtils::create<String>(db_struct_));
+  ams_param.setValue("positive_adducts", positive_adducts_);
+  ams_param.setValue("negative_adducts", negative_adducts_);
+
+  AccurateMassSearchEngine ams;
+  ams.setParameters(ams_param);
+  ams.init();
+
+  ams.run(input, output);
+}
+
+MSSpectrum FIAMSDataProcessor::trackNoise(const MSSpectrum & input) {
+  SignalToNoiseEstimatorMedianRapid sne(50);
+  MSSpectrum output;
+  if (input.size() == 0) {
+    return output;
+  }
+  std::vector<double> mzs, intensities;
+  for (auto it = input.begin(); it != input.end(); ++it)
+  {
+      mzs.push_back(it->getMZ());
+      intensities.push_back(it->getIntensity());
+  }
+  SignalToNoiseEstimatorMedianRapid::NoiseEstimator e = sne.estimateNoise(mzs, intensities);
+  
+  for (auto it = input.begin(); it != input.end(); ++it)
+  {
+      Peak1D peak;
+      peak.setMZ(it->getMZ());
+      peak.setIntensity(e.get_noise_value(it->getMZ()));
+      output.push_back(peak);
+  }
+  return output;
+}
+
+void FIAMSDataProcessor::run(
+  float n_seconds,
+  OpenMS::MzTab & output
+  ) {
+  vector<MSSpectrum> output_cut;
+  cutForTime(experiment_, output_cut, n_seconds);
+  MSSpectrum merged_spectrum = mergeAlongTime(output_cut);
+  MSSpectrum picked_spectrum = extractPeaks(merged_spectrum);
+  String postfix = String(static_cast<int>(n_seconds));
+  if (store_progress_) {
+    storeSpectrum_(merged_spectrum, dir_output_ + "/" + filename_ + "_merged_" + postfix + ".mzML");
+    storeSpectrum_(picked_spectrum, dir_output_ + "/" + filename_ + "_picked_" + postfix + ".mzML");
+  }
+  FeatureMap picked_features = convertToFeatureMap(picked_spectrum);
+  MSSpectrum signal_to_noise = trackNoise(picked_spectrum);
+  storeSpectrum_(signal_to_noise, dir_output_ + "/" + filename_ + "_signal_to_noise_" + postfix + ".mzML");
+  runAccurateMassSearch(picked_features, output);
+  OpenMS::MzTabFile mztab_outfile;
+  mztab_outfile.store(dir_output_ + "/" + filename_ + "_" + postfix + ".mzTab", output);
+}
+
+void FIAMSDataProcessor::loadExperiment_() {
```

Comment:
`loadExperiment_` is a bit out of scope for the class.  I think it would be better if the user calls those snippets of code prior to calling the method.